rest-service/manager_rest/rest/resources_v3_1/execution_schedules.py

```diff
@@ -0,0 +1,275 @@
+            'force': verify_and_convert_bool(
+                'force',
+                request_dict.get('force', False)),
+            'dry_run': verify_and_convert_bool(
+                'dry_run',
+                request_dict.get('dry_run', False)),
+            'queue': verify_and_convert_bool(
+                'queue',
+                request_dict.get('queue', False)),
+            'wait_after_fail': request_dict.get('wait_after_fail', 600)
+        }
+        parameters = request_dict.get('parameters', None)
+        if parameters is not None and parameters.__class__ is not dict:
+            raise manager_exceptions.BadParametersError(
+                "request body's 'parameters' field must be a dict but"
+                " is of type {0}".format(parameters.__class__.__name__))
+
+        # --- Handle deployment
+        rm = get_resource_manager()
+        deployment_id = request_dict['deployment_id']
+        deployment = rm.sm.get(models.Deployment, deployment_id)
+        rm._validate_permitted_to_execute_global_workflow(deployment)
+        rm._verify_workflow_in_deployment(workflow_id,
+                                          deployment,
+                                          deployment_id)
+
+        # --- Schedule event parameters
+        since = self._verify_timestamp(request_dict['since'])
+        until = self._verify_timestamp(request_dict.get('until'))
+
+        # we need to have at least: rrule; or count=1; or frequency
+        rrule_string = request_dict.get('rrule')
+        frequency = request_dict.get('frequency')
+        weekdays = request_dict.get('weekdays')
+        count = request_dict.get('count')
+
+        if rrule_string:
+            if frequency or weekdays or count:
+                raise manager_exceptions.BadParametersError(
+                    "`rrule` cannot be provided together with `frequency`, "
+                    "`weekdays` or `count`.")
+            try:
+                rrule.rrulestr(rrule_string)
+            except ValueError as e:
+                raise manager_exceptions.BadParametersError(
+                    "invalid RRULE string provided: {}".format(str(e)))
+            rule = {'rrule': rrule_string}
+        else:
+            if count:
+                count = convert_to_int(request_dict.get('count'))
+            frequency = self._verify_frequency(request_dict.get('frequency'))
+            weekdays = self._verify_weekdays(request_dict.get('weekdays'))
+            if not frequency and count != 1:
+                raise manager_exceptions.BadParametersError(
+                    "frequency must be specified for execution count larger "
+                    "than 1")
+            rule = {
+                'frequency': frequency,
+                'count': count,
+                'weekdays': weekdays
+            }
+
+        slip = request_dict['slip']
+        stop_on_fail = verify_and_convert_bool(
+            'stop_on_fail',  request_dict.get('stop_on_fail', False))
```

Comment:
figuring out the rules can fit well in a separate method as well, and it would be easy to test just that method too